pkg/aws/aws.go

```diff
@@ -248,18 +237,35 @@ func (a *AWS) Collect(ch chan<- prometheus.Metric) {
 	providerScrapesTotalCounter.WithLabelValues(subsystem).Inc()
 }
 
-func newEc2Client(region, profile string) (*ec2.Client, error) {
-	options := []func(*awsconfig.LoadOptions) error{awsconfig.WithEC2IMDSRegion()}
-	options = append(options, awsconfig.WithRegion(region))
-	if profile != "" {
-		options = append(options, awsconfig.WithSharedConfigProfile(profile))
+func newEc2Client(region string, ac aws.Config) (*ec2.Client, error) {
+	ac, err := awsconfig.LoadDefaultConfig(
+		context.Background(),
+		awsconfig.WithRegion(region),
+	)
+	if err != nil {
+		return nil, err
 	}
-	// Set max retries to 10. Throttling is possible after fetching the pricing data, so setting it to 10 ensures the next scrape will be successful.
-	options = append(options, awsconfig.WithRetryMaxAttempts(maxRetryAttempts))
+
+	return ec2.NewFromConfig(ac), nil
+}
+
+func assumeRole(roleARN string, options []func(*awsconfig.LoadOptions) error) ([]func(*awsconfig.LoadOptions) error, error) {
```

Comment:
This is not a full refactor but I'm splitting this as a helper func since the plan is to refactor `New` at some point...